tower-http/src/services/fs/serve_dir.rs

```diff
@@ -92,9 +92,53 @@ use tower_service::Service;
 #[derive(Clone, Debug)]
 pub struct ServeDir {
     base: PathBuf,
-    append_index_html_on_directories: bool,
     buf_chunk_size: usize,
     precompressed_variants: Option<PrecompressedVariants>,
+    // This is used to specialise implementation for
+    // single files
+    variant: ServeVariant,
+}
+
+// Allow the ServeDir service to be used in the ServeFile service
+// with almost no overhead
+#[derive(Clone, Debug)]
+enum ServeVariant {
+    Directory {
+        append_index_html_on_directories: bool,
+    },
+    SingleFile {
+        mime: HeaderValue,
+    },
+}
```

Comment:
This is a great idea! Much simpler than the design I had in mind 👍

Reply:
Thanks!